devices.go

```diff
@@ -146,29 +153,57 @@ func (d *Device) GetPortByName(name string) (*Port, error) {
 	return nil, fmt.Errorf("Port %s not found in device %s", name, d.ID)
 }
 
-func (d *Device) GetNetworkType() (string, error) {
-	numOfBonds := d.NumOfBonds()
-	if (numOfBonds < 1) || (numOfBonds > 2) {
-		return "", fmt.Errorf("Wrong number of Bond ports")
+type ports map[string]*Port
+
+func (ports ports) allBonded() bool {
+	if ports == nil {
+		return false
 	}
-	bond0, err := d.GetPortByName("bond0")
-	if err != nil {
-		return "", err
+
+	if len(ports) == 0 {
+		return false
 	}
-	if numOfBonds == 2 {
-		bond1, err := d.GetPortByName("bond1")
-		if err != nil {
-			return "", err
+
+	for _, p := range ports {
+		if (p == nil) || (!p.Data.Bonded) {
+			return false
 		}
-		if bond0.NetworkType == bond1.NetworkType {
-			return bond0.NetworkType, nil
+	}
+	return true
+}
+
+func (d *Device) HasManagementIPs() bool {
+	for _, ip := range d.Network {
+		if ip.Management {
+			return true
+		}
+	}
+	return false
+}
+
+func (d *Device) GetNetworkType() (string, error) {
```

Comment:
Maybe remove the error return value (since it's always nil). And fix it in all the calls.

Reply:
I was trying to avoid breaking the packngo API with this change. I was also not sure if I would need to reintroduce some errors during the review.

I'll update this (if there are definitely no error conditions we are missing) and the changelog and leave a note about the API stability in packngo while the project is pre-v1.

Reply:
I wouldn't worry about this too much. This is golang, once you update packngo as a dependency, you will immediately see that the return value type change. It's an easy fix.

IMO better to slightly break the API than returning an err that's never set.